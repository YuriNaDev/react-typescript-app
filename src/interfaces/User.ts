export interface IUser {
	id: number
	first_name: string
	last_name: string
	email: string
}

export interface IUserResponse {
	data: IUser[]
}
