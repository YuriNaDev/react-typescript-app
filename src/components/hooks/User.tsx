import React, { useState } from 'react'
import { Typography, Box, Button } from '@material-ui/core'
import { IUser } from 'interfaces/User'

const User: React.FC = () => {
	const [user, setUser] = useState<IUser | null>(null)

	const toggleUser = () =>
		setUser(state =>
			state
				? null
				: {
						id: 0,
						first_name: 'Leanne',
						last_name: 'Graham',
						email: 'Sincere@april.biz',
				  }
		)

	return (
		<Box py={2}>
			<Typography variant="h6" gutterBottom>
				{user ? (
					<>
						name: {user.first_name} {user.last_name}
						<br />
						email: {user.email}
					</>
				) : (
					'사용자 알 수 없음'
				)}
			</Typography>
			<Button color="primary" variant="outlined" onClick={toggleUser}>
				TOGGLE
			</Button>
		</Box>
	)
}

export default User
