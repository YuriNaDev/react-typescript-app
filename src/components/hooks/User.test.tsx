import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import User from './User'

describe('<User />', () => {
	test('matches snapshot', () => {
		const { container } = render(<User />)
		expect(container).toMatchSnapshot()
	})

	test('use toggle button', () => {
		const { getByText } = render(<User />)

		const typography = getByText('사용자 알 수 없음')
		const button = getByText('TOGGLE')

		fireEvent.click(button)
		expect(typography).not.toHaveTextContent('사용자 알 수 없음')

		fireEvent.click(button)
		expect(typography).toHaveTextContent('사용자 알 수 없음')
	})
})
