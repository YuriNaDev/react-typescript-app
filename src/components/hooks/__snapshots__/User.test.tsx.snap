// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
    >
      사용자 알 수 없음
    </h6>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        TOGGLE
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
